src/interfaces/wallet.cpp

```diff
@@ -173,6 +173,7 @@ class CoinJoinImpl : public CoinJoin::Client
 class WalletImpl : public Wallet
 {
 public:
+    unordered_lru_cache<uint256, shared_ptr<WalletTx>, std::hash<uint256>, 10000> m_WalletTxCache;
```

Comment:
Creates the cache of WalletTx.  These were being recreated continuously and they don't change.  With a lot of vins/vouts, it was rebuilding vectors every call and all of this too a lot of time.  Additionally, with many vouts this creation was being done for each vout for the same Tx.